File: leetcode/heap/0621_task_scheduler.cpp
// LC 621. Task Scheduler | Medium
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int leastInterval(vector<char>& tasks, int n) {
        int freq[26] = {};
        for (char c : tasks) freq[c-'A']++;
        int maxF = *max_element(freq, freq+26);
        int maxCnt = count(freq, freq+26, maxF);
        return max((int)tasks.size(), (maxF-1)*(n+1)+maxCnt);
    }
};

File: leetcode/heap/0621_task_scheduler.py
# LC 621. Task Scheduler | Medium
from collections import Counter

class Solution:
    def leastInterval(self, tasks, n):
        freq = list(Counter(tasks).values())
        max_freq = max(freq)
        max_count = freq.count(max_freq)
        return max(len(tasks), (max_freq - 1) * (n + 1) + max_count)
